magda-sleuther-framework/src/index.ts

```diff
@@ -9,10 +9,26 @@ import AsyncPage, {
 } from "@magda/typescript-common/dist/AsyncPage";
 import SleutherOptions from "./SleutherOptions";
 import setupWebhookEndpoint from "./setupWebhookEndpoint";
+import setupRecrawlEndpoint from "./setupRecrawlEndpoint";
+import startApiEndpoints from "./startApiEndpoints";
 import isWebhookRegistered from "./isWebhookRegistered";
 import registerWebhook from "./registerWebhook";
 import resumeWebhook from "./resumeWebhook";
 
+let isCrawling: boolean = false;
+let crawlingPageToken: string = "";
+let crawledRecordNumber: number = 0;
```

Comment:
Could we lift all this crawler state stuff into its own class instead of having it in `index.ts` ?